XenAdmin/Controls/NetworkingTab/NetworkList.cs

```diff
@@ -732,31 +758,40 @@ private void launchVmNetworkSettingsDialog()
                 return;
 
             int device;
-            VIFDialog d;
-            if (int.TryParse(vif.device, out device))
-            {
-                d = new VIFDialog(vm.Connection, vif, device, vm.HasSriovRecommendation());
-            }
-            else
+            if (!int.TryParse(vif.device, out device))
             {
                 log.ErrorFormat("Aborting vif edit. Could not parse existing vif device to int. Value is: '{0}'", vif.device);
                 return;
             }
 
-            if (d.ShowDialog() != DialogResult.OK)
-                return;
+            VIF proxyVIF;
+            using (var d = new VIFDialog(vm.Connection, vif, device, vm.HasSriovRecommendation()))
+            {
+                if (d.ShowDialog() != DialogResult.OK)
+                    return;
+
+                proxyVIF = d.GetNewSettings();
+            }
 
-            var proxyVIF = d.GetNewSettings();
-            UpdateVIFCommand command = new UpdateVIFCommand(Program.MainWindow, vm, vif, proxyVIF);
+            var action = new UpdateVIFAction(vm, vif, proxyVIF);
+            action.Completed += updateVIFAction_Completed;
             InBuildList = true;
-            command.Completed += new EventHandler((s, f) => Program.Invoke(this, () =>
-                                                                                     {
-                                                                                         InBuildList = false;
-                                                                                         BuildList();
-                                                                                     }));
-            command.Execute();
+            action.RunAsync();
         }
 
+        private void updateVIFAction_Completed(ActionBase sender)
+        {
+            sender.Completed -= updateVIFAction_Completed;
+            var action = (AsyncAction)sender;
+
+            Program.Invoke(Program.MainWindow, () =>
+            {
+                if (action.Result == false.ToString())

```

Comment:
I think we should show the hotplug error after the page is updated (as per old behaviour)